samples/sampler/stories/qa/VirtualGridList.js

```diff
@@ -1,180 +1,203 @@
+		closePopup();
+	}, [closePopup]);
+
+	const componentRenderItem = useCallback(({index, ...rest}) => {
+		return (
+			/* eslint-disable react/jsx-no-bind */
+			<Item key={index} onClick={() => componentClosePopup(index)} {...rest}>{itemList[index]}</Item>
+		);
+	}, [componentClosePopup]);
+
+	const getScrollTo = useCallback((newScrollTo) => {
+		scrollTo.current = newScrollTo;
+	}, []);
+
+	return (
+		<div {...props} style={{width: ri.scaleToRem(915), height: ri.scaleToRem(600)}}>
+			<VirtualGridList
+				cbScrollTo={getScrollTo}
+				dataSize={itemList.length}
+				direction="vertical"
+				itemRenderer={componentRenderItem}
+				itemSize={{minWidth: ri.scale(285), minHeight: ri.scale(60)}}
+			/>
+		</div>
+	);
+};
+
+MyVirtualList.propTypes = {
+	closePopup: PropTypes.func
+};
+
+const ButtonAndVirtualGridList = () => {
+	const [isPopup, setIsPopup] = useState(false);
+
+	const openPopup = useCallback(() => {
+		setIsPopup(true);
+	}, []);
+
+	const closePopup = useCallback(() => {
+		setIsPopup(false);
+	}, []);
+
+	const renderPopup = useCallback((rest) => {
+		return (
+			<MyVirtualList {...rest} closePopup={closePopup} />
+		);
+	}, [closePopup]);
+
+	return (
+		<div>
+			<ContextualPopupButton
+				direction="right"
+				onClick={openPopup}
+				onClose={closePopup}
+				open={isPopup}
+				popupComponent={renderPopup}
+				showCloseButton
+				spotlightRestrict="self-only"
+			>
+				Focus here
+			</ContextualPopupButton>
+		</div>
+	);
+};
+
+export default {
+	title: 'Moonstone/VirtualGridList',
+	component: 'VirtualGridList'
+};
+
+export const HorizontalVirtualGridList = (args) => (
+	<VirtualGridList
+		dataSize={updateDataSize(args['dataSize'])}
+		direction="horizontal"
+		focusableScrollbar={args['focusableScrollbar']}
+		horizontalScrollbar={args['horizontalScrollbar']}
+		itemRenderer={renderItem}
+		itemSize={{
+			minWidth: ri.scale(args['minWidth']),
+			minHeight: ri.scale(args['minHeight'])
+		}}
+		noScrollByWheel={args['noScrollByWheel']}
+		onKeyDown={action('onKeyDown')}
+		onScrollStart={action('onScrollStart')}
+		onScrollStop={action('onScrollStop')}
+		spacing={ri.scale(args['spacing'])}
+		spotlightDisabled={args['spotlightDisabled']}
+		verticalScrollbar={args['verticalScrollbar']}
+		wrap={args['wrap']}
+	/>
+);
+
+number('dataSize', HorizontalVirtualGridList, Config, defaultDataSize);
+boolean('focusableScrollbar', HorizontalVirtualGridList, Config);
+select('horizontalScrollbar', HorizontalVirtualGridList, prop.scrollbarOption, Config);
+number('minWidth', HorizontalVirtualGridList, Config, 180);
+number('minHeight', HorizontalVirtualGridList, Config, 270);
+boolean('noScrollByWheel', HorizontalVirtualGridList, Config);
+number('spacing', HorizontalVirtualGridList, Config, 18);
+boolean('spotlightDisabled', HorizontalVirtualGridList, Config, false);
+select('verticalScrollbar', HorizontalVirtualGridList, prop.scrollbarOption, Config);
+select('wrap', HorizontalVirtualGridList, wrapOption, Config);
```

Comment:
In virtualList and virtualListNative we use ['false', 'true', '"noAnimation"'] instead of wrapOption to avoid showing in controls the option &quot;noAnimation&quot;. Maybe we should do the same here.

Reply:
actually the virtualgridlist approach is correct. in virtualLit and virtualListNative  is not correct